samples/net/wifi/wifi_station/src/main.c

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2022 Espressif Systems (Shanghai) Co., Ltd.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/zephyr.h>
+#include <errno.h>
+#include <zephyr/net/net_if.h>
+#include <zephyr/net/net_core.h>
+#include <zephyr/net/net_context.h>
+#include <zephyr/net/net_mgmt.h>
+#include <zephyr/net/wifi_mgmt.h>
+#include <zephyr/net/net_conn_mgr.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(wifi_station, LOG_LEVEL_DBG);
+
+#define WIFI_MASK (NET_EVENT_WIFI_CONNECT_RESULT | NET_EVENT_WIFI_DISCONNECT_RESULT)
+
+static struct net_mgmt_event_callback wifi_mgmt_cb;
+static int connected;
+
+static void wifi_event_handler(struct net_mgmt_event_callback *cb, uint32_t mgmt_event,
+			       struct net_if *iface)
+{
+	const struct wifi_status *status = (const struct wifi_status *)cb->info;
+
+	switch (mgmt_event) {
+	case NET_EVENT_WIFI_CONNECT_RESULT:
+		if (status->status) {
+			LOG_ERR("Connection request failed (%d)", status->status);
+		} else {
+			LOG_INF("WiFi connected");
+			connected = 1;
+		}
+		break;
+	case NET_EVENT_WIFI_DISCONNECT_RESULT:
+		LOG_INF("WiFi disconnected");
+		break;
+	default:
+		break;
+	}
+}
+
+void main(void)
+{
+	int ret = 0;
+
+	net_mgmt_init_event_callback(&wifi_mgmt_cb, wifi_event_handler, WIFI_MASK);
+	net_mgmt_add_event_callback(&wifi_mgmt_cb);
+
+	struct net_if *iface = net_if_get_default();
+	static struct wifi_connect_req_params cnx_params = {
+		.ssid = CONFIG_WIFI_SSID,
+		.psk = CONFIG_WIFI_PSK,
+		.channel = 0,
+		.security = WIFI_SECURITY_TYPE_PSK,
+	};
+
+	cnx_params.ssid_length = strlen(CONFIG_WIFI_SSID);
+	cnx_params.psk_length = strlen(CONFIG_WIFI_PSK);
+
+	LOG_INF("connecting to %s...", CONFIG_WIFI_SSID);
```

Comment:
Not sure what is wrong, but I couldn't connect to any of my 2 APs that I have at home. It just hangs at:
```
[00:00:00.826,000] <inf> wifi_station: connecting to AX3600...
```
Didn't try to debug yet.